src/analysis/calculus/fderiv.lean

```diff
@@ -1914,6 +1914,59 @@ h.continuous.comp (continuous_const.prod_mk continuous_id)
 
 end bilinear_map
 
+namespace continuous_linear_equiv
+
+/-!
+### The set of continuous linear equivalences between two Banach spaces is open
+
+In this section we establish that the set of continuous linear equivalences between two Banach
+spaces is an open subset of the space of linear maps between them.  These facts are placed here
+because the proof uses `is_bounded_bilinear_map_comp.continuous`, proved just above as a consequence
+of its differentiability.
+-/
+
+lemma is_open [complete_space E] :
```

Comment:
```suggestion
protected lemma is_open [complete_space E] :
```

Reply:
Could you explain this change?  IIUC `protected` is for things that are not generally useful, but this fact does seem generally useful

Reply:
> IIUC `protected` is for things that are not generally useful, but this fact does seem generally useful

I think you might be confusing `private` and `protected`. Briefly,
- `private` makes it so that you can't refer at all to a declaration (except by trickery) outside the section or file; I think we mostly mark things `private` in mathlib when they're followed by something strictly more general.
- `protected` makes it so that the declaration can only be referred to in fully-qualified form ([obligatory TPiL link](https://leanprover.github.io/theorem_proving_in_lean/interacting_with_lean.html#more-on-namespaces)). In this case, even if we `open continuous_linear_equiv`, we still have to refer to this lemma using `continuous_linear_equiv.is_open`. Marking things `protected` is useful when there are lots of things with the same name under different namespaces.

Reply:
Ah!  Thanks.